src/main/java/com/github/junrar/rarfile/FileHeader.java

```diff
@@ -186,41 +200,129 @@ public FileHeader(BlockHeader bh, byte[] fileHeader) {
         }
-        mTime = getDateDos(fileTime);
-        // TODO rartime -> extended
 
+        setMTime(getDateDos(fileTime));
+
+        if (hasExtTime()) {
+            short extTimeFlags = Raw.readShortLittleEndian(fileHeader, position);
+            position += 2;
+
+            {
+                int flag = extTimeFlags >>> 12;
+                if ((flag & 0x8) != 0) {
+                    int count = flag & 0x3;
+                    long remainder = 0;
+                    for (int i = 0; i < count; i++) {
+                        int b = fileHeader[position] & 0xff;
+                        remainder = (b << 16) | (remainder >>> 8);
+                        position++;
+                    }
+                    long micros = remainder / MICROS_PER_UNIT;
+                    if ((flag & 0x4) != 0) {
+                        micros += MICROS_PER_SECOND;
+                    }
+                    setLastModifiedTime(FileTime.from(Instant.ofEpochMilli(mTime.getTime()).plus(micros, ChronoUnit.MICROS)));
+                }
+            }
+
+            {
+                int flag = extTimeFlags >>> 8;
+                if ((flag & 0x8) != 0) {
+                    cTime = getDateDos(Raw.readIntLittleEndian(fileHeader, position));
+                    position += 4;
+                    int count = flag & 0x3;
+                    long remainder = 0;
+                    for (int i = 0; i < count; i++) {
+                        int b = fileHeader[position] & 0xff;
+                        remainder = (b << 16) | (remainder >>> 8);
+                        position++;
+                    }
+                    long micros = remainder / MICROS_PER_UNIT;
+                    if ((flag & 0x4) != 0) {
+                        micros += MICROS_PER_SECOND;
+                    }
+                    setCreationTime(FileTime.from(Instant.ofEpochMilli(cTime.getTime()).plus(micros, ChronoUnit.MICROS)));
+                }
+            }
+
+            {
+                int flag = extTimeFlags >>> 4;
+                if ((flag & 0x8) != 0) {
+                    aTime = getDateDos(Raw.readIntLittleEndian(fileHeader, position));
+                    position += 4;
+                    int count = flag & 0x3;
+                    long remainder = 0;
+                    for (int i = 0; i < count; i++) {
+                        int b = fileHeader[position] & 0xff;
+                        remainder = (b << 16) | (remainder >>> 8);
+                        position++;
+                    }
+                    long micros = remainder / MICROS_PER_UNIT;
+                    if ((flag & 0x4) != 0) {
+                        micros += MICROS_PER_SECOND;
+                    }
+                    setLastAccessTime(FileTime.from(Instant.ofEpochMilli(aTime.getTime()).plus(micros, ChronoUnit.MICROS)));
+                }
+            }
+
+            {
+                //noinspection PointlessBitwiseExpression
+                int flag = extTimeFlags >>> 0;
```

Comment:
@gotson This is here just we I can reuse the block below without changing anything. I figured out how to make it into a function and I'm gonna refactor it so it'll be a lot cleaner.